### ISSUE TITLE
New User Role Assignment doesn't stick upon initial User creation

### ISSUE THREAD

**Issue by kishanjasani** [buddyboss/buddyboss-platform#2921]
**Describe the bug**
The New user Role-Assignments not being honored by the WP Backend System. It is not a third-party plugin or theme issue. The steps/actions that led to this issue are listed below. I have taken time and effort to distill the steps down to as few as possible.

**To Reproduce**
1. Given a basic BuddyBoss setup—with both mandatory plugins and theme
active—add a couple of basic, manually-created test users for good measure,
as well as to test and see (first-hand) that the system works as expected.

2. Go to BuddyBoss > Components and enable all components

3. Go to BuddyBoss > Settings > Profiles > Profile Types

   - a. select “Enable profile types to give members unique profile fields and
permissions”

   - b. click [Save Settings]

4. Go to BuddyBoss > Settings > Groups > Group Types

   - a. select “Enable group types to better organize groups enabled”

   - b. click [Save Settings]

5. Add a couple of interim/test Profile Types

   - a. BuddyBoss > Profile > Profile Types [Add New]

   - b. Model your test Profile Types according to the ScreenShot ()

   - c. Click [Publish]

6. Go to BuddyBoss> Settings > Profiles > Profile Types:

   - a. From the Default Profile Type Dropdown, select “ProfileType-01s”*

   - b. Click [Save Settings]*

7. Create a new member by going to: WP > Users > Add New

   - a. Enter desired test user data

   - b. select any role other than Subscriber”

   - c. click [Add New User]

**Screenshots**
https://www.youtube.com/watch?v=84soWG78UZM

**Support ticket links**
https://secure.helpscout.net/conversation/1618716025/161848?folderId=3709082

Issue: #2140 

**Jira issue** : [PROD-740]

[PROD-740]: https://buddyboss.atlassian.net/browse/PROD-740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by chrisdubia** [comment]
We are also experiencing this issue:

Profile type not correctly being assigned when inviting to site

We implement the profile type option in buddy boss and created three 4 types

Administrator
Teacher
Student
Guest

When using Send Invite via the front end of the site our Administrators (Buddyboss profile type) are the only ones that can invite and also can choose to assign Buddyboss profile type upon invite.

Buddyboss Administrators are able to send the invite, but the profile type does not correctly get assigned.  The profile type that is getting assigned is Member - which is not a Profile type that we created.

We have tested this on Staging as well with no plugins activated or custom code.

To replicate this:

Login to your Sandbox or site

Go you your profile image in the upper right corner
Send Invite - make sure to choose a profile type
Accept the invite and register as the new user
Login as the  new user
You will see that the profile type that is assigned it Member

When is this going to be fixed?
